tests/test_es_index_manager.py

```diff
@@ -0,0 +1,242 @@
+"""Testing for Elasticsearch Index Manager."""
```

Comment:
I have some suggested changes for this file which allows the tests to pass and cleans up the fixtures:
- Removed unused code
- Passes in config instead of ConfigEntry where required
- Passes in ConfigEntry instead of config where required
- Adjusts test fixtures to shutdown the gateway

```diff
diff --git a/tests/test_es_index_manager.py b/tests/test_es_index_manager.py
index a0d9b21..3f7ba52 100644
--- a/tests/test_es_index_manager.py
+++ b/tests/test_es_index_manager.py
@@ -1,9 +1,9 @@
 """Testing for Elasticsearch Index Manager."""
 
 import pytest
+from elasticsearch.utils import get_merged_config
 from elasticsearch7.exceptions import ElasticsearchException
 from homeassistant.core import HomeAssistant
-from homeassistant.setup import async_setup_component
 from pytest_homeassistant_custom_component.common import MockConfigEntry
 from pytest_homeassistant_custom_component.test_util.aiohttp import AiohttpClientMocker
 
@@ -19,7 +19,6 @@ from custom_components.elasticsearch.const import (
 from custom_components.elasticsearch.errors import ElasticException
 from custom_components.elasticsearch.es_gateway import ElasticsearchGateway
 from custom_components.elasticsearch.es_index_manager import IndexManager
-from tests.conftest import mock_config_entry
 from tests.test_util.aioclient_mock_utils import (
     extract_es_ilm_template_requests,
     extract_es_legacy_index_template_requests,
@@ -28,18 +27,6 @@ from tests.test_util.aioclient_mock_utils import (
 from tests.test_util.es_startup_mocks import mock_es_initialization
 
 
-async def _setup_config_entry(hass: HomeAssistant, mock_entry: mock_config_entry):
-    mock_entry.add_to_hass(hass)
-    assert await async_setup_component(hass, DOMAIN, {}) is True
-    await hass.async_block_till_done()
-
-    config_entries = hass.config_entries.async_entries(DOMAIN)
-    assert len(config_entries) == 1
-    entry = config_entries[0]
-
-    return entry
-
-
 @pytest.fixture()
 async def legacy_index_manager(
     hass: HomeAssistant, es_aioclient_mock: AiohttpClientMocker
@@ -60,15 +47,17 @@ async def legacy_index_manager(
         title="ES Config",
     )
 
-    entry = mock_entry  # entry = await _setup_config_entry(hass, mock_entry)
+    entry = mock_entry
 
-    gateway = ElasticsearchGateway(entry)
+    gateway = ElasticsearchGateway(config_entry=entry)
 
     await gateway.async_init()
 
-    index_manager = IndexManager(hass, entry, gateway)
+    index_manager = IndexManager(hass, get_merged_config(entry), gateway)
+
+    yield index_manager
 
-    return index_manager
+    await gateway.async_stop_gateway()
 
 
 @pytest.fixture()
@@ -99,18 +88,20 @@ async def modern_index_manager(
 
     entry = mock_entry  # entry = await _setup_config_entry(hass, mock_entry)
 
-    gateway = ElasticsearchGateway(entry)
+    gateway = ElasticsearchGateway(config_entry=entry)
 
     await gateway.async_init()
 
-    index_manager = IndexManager(hass, entry, gateway)
+    index_manager = IndexManager(hass, get_merged_config(entry), gateway)
+
+    yield index_manager
 
-    return index_manager
+    await gateway.async_stop_gateway()
 
 
 @pytest.mark.asyncio
 async def test_legacy_index_mode_setup(
-    legacy_index_manager: legacy_index_manager, es_aioclient_mock: AiohttpClientMocker
+    legacy_index_manager: IndexManager, es_aioclient_mock: AiohttpClientMocker
 ):
     """Test for legacy index mode setup."""
     legacy_template_requests = extract_es_legacy_index_template_requests(
@@ -138,7 +129,7 @@ async def test_legacy_index_mode_setup(
 
 @pytest.mark.asyncio
 async def test_modern_index_mode_setup(
-    modern_index_manager: modern_index_manager, es_aioclient_mock: AiohttpClientMocker
+    modern_index_manager: IndexManager, es_aioclient_mock: AiohttpClientMocker
 ):
     """Test for modern index mode setup."""
 
@@ -193,14 +184,13 @@ async def test_invalid_index_mode_setup(
         title="ES Config",
     )
 
-    entry = mock_entry  # entry = await _setup_config_entry(hass, mock_entry)
-
-    gateway = ElasticsearchGateway(entry)
+    entry = mock_entry
+    gateway = ElasticsearchGateway(config_entry=entry)
 
     await gateway.async_init()
 
     with pytest.raises(ElasticException):
-        indexmanager = IndexManager(hass, entry, gateway)
+        indexmanager = IndexManager(hass, get_merged_config(entry), gateway)
         await indexmanager.async_setup()
 
     await gateway.async_stop_gateway()
@@ -229,14 +219,14 @@ async def test_invalid_legacy_with_serverless(
         title="ES Config",
     )
 
-    entry = mock_entry  # entry = await _setup_config_entry(hass, mock_entry)
+    entry = mock_entry
 
-    gateway = ElasticsearchGateway(entry)
+    gateway = ElasticsearchGateway(config_entry=entry)
 
     await gateway.async_init()
 
     with pytest.raises(ElasticsearchException):
-        indexmanager = IndexManager(hass, entry, gateway)
+        indexmanager = IndexManager(hass, get_merged_config(entry), gateway)
         await indexmanager.async_setup()
 
     await gateway.async_stop_gateway()
```

Reply:
yeah sorry i was working on refactoring this one this morning and hadn't pushed a commit in a minute. I'll incorporate these changes and push my update to these tests shortly